### PR TITLE
fix(logging): improve go core's debug logging

### PR DESCRIPTION
**Pull request by padamstx**
This commit adds some debug logging in various
components, and also modifies some tests so that
it's easy to enable debug logging when running
individual sets of tests.

### CODE REVIEW

File: core/config_utils_test.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+var (
+	// To enable debug logging during test execution, set this to "LevelDebug"
+	configUtilsTestLogLevel LogLevel = LevelDebug
```
Comment (pyrooka) [inline core/config_utils_test.go:30]: Shouldn't this be configured to error level?

File: core/base_service.go [hunk 2]
```diff
@@ -826,6 +834,16 @@ var (
...
+
+	// A regular expression to match the error returned by net/http when a
+	// request header or value is invalid. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	invalidHeaderErrorRe = regexp.MustCompile(`invalid header`)
+
+	// A regular expression to match the error returned by net/http when the
+	// TLS certificate is not trusted. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	notTrustedErrorRe = regexp.MustCompile(`certificate is not trusted`)
```
Comment (pyrooka) [inline core/base_service.go:846]: You've probably made a more thorough investigation than me - just right now, but according to the following code snippet from Go's source code, we might need to add a Windows specific regex pattern to make sure this package is truly cross-platform.
```go
macosErr:   "x509: “valid.testing.golang.invalid” certificate is not trusted",
windowsErr: "x509: certificate signed by unknown authority",
```
[source](https://github.com/golang/go/blob/aeac0b6cbfb42bc9c9301913a191bb09454d316a/src/crypto/x509/platform_test.go#L171)

**Review by pyrooka — APPROVED** [review]
I left 2 minor suggestions, otherwise it looks pretty good!

File: core/base_service.go (hunk 2 above)
Comment (padamstx) [reply core/base_service.go:846]: When I originally wrote the `IBMCloudSDKRetryPolicy()` function, I essentially stole most of the code from the `go-retryablehttp.ErrorPropagatedRetryPolicy`, located [here](https://github.com/hashicorp/go-retryablehttp/blob/40b0cad1633fd521cee5884724fcf03d039aaf3f/client.go#L486).

It looks like you referenced a section of a testcase.  Are we sure that on windows we do in fact get a different error message?

Comment (pyrooka) [reply core/base_service.go]: I resolve this thread as we discussed it on Slack. Phil suggested to use
```go
notTrustedErrorRe = regexp.MustCompile(`certificate is not trusted|certificate is signed by unknown authority`)
```
and I think that's a good idea. It should cover most if not all situations.

File: core/config_utils_test.go (hunk 1 above)
Comment (padamstx) [reply core/config_utils_test.go]: yes... yes it should :joy:

Comment (padamstx) [reply core/config_utils_test.go]: fixed it

File: core/base_service.go (hunk 2 above)
Comment (padamstx) [reply core/base_service.go]: I updated the regex to include both substrings.

File: core/base_service.go [hunk 3]
```diff
@@ -289,6 +291,7 @@ func (service *BaseService) DisableSSLVerification() {
 		// Disable server ssl cert & hostname verification.
 		tr.TLSClientConfig.InsecureSkipVerify = true // #nosec G402
 	}
+	GetLogger().Debug("Disabled SSL verification in HTTP client")
```
Comment (dpopp07) [inline core/base_service.go:294]: This message seems helpful but I don't think I remember seeing this in the other cores (or at least, not in Node). Perhaps we should add it there as well? (Maybe I missed it)

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: core/base_service.go (hunk 3 above)
Comment (padamstx) [reply core/base_service.go:294]: yeah, it turns out that we have no `disableSslVerification()` in the node core's BaseService class.

Comment (dpopp07) [reply core/base_service.go:294]: Oh, interesting. I think there is in the authenticators in the Node core. Odd that we aren't consistent there. Perhaps we should be, but that can be a follow-on effort.

**Pull request merged**